tutorials/cyber/network_mapping/lanl_network_mapping_using_rapids.ipynb

```diff
@@ -0,0 +1,781 @@
+    "    log_df_processed = log_df_processed[:0]\n",
+    "    \n",
+    "    columns = list(regexes[eventcode].keys())\n",
+    "\n",
+    "    for col in columns:\n",
+    "        regex_pattern = regexes[eventcode].get(col)\n",
+    "        extracted_nvstrings = df['Raw'].str.extract(regex_pattern)\n",
+    "        \n",
+    "        if not extracted_nvstrings.empty:\n",
+    "            log_df_processed[col] = extracted_nvstrings[0]\n",
+    "    \n",
+    "    for col in log_df_processed.columns:\n",
+    "        if not log_df_processed[col].empty:\n",
+    "            if log_df_processed[col].dtype == 'float64':\n",
+    "                log_df_processed[col] = log_df_processed[col].astype('int').astype('str')\n",
+    "            elif log_df_processed[col].dtype == 'object':\n",
+    "                pass\n",
+    "            \n",
+    "            else:\n",
+    "                log_df_processed[col] = log_df_processed[col].astype('str')\n",
+    "        \n",
+    "        if log_df_processed[col].empty:\n",
+    "            log_df_processed[col] = nvstrings.to_device([])\n",
+    "    \n",
+    "    return log_df_processed\n",
+    "\n",
+    "\n",
+    "def filter_by_pattern(df, pattern):\n",
+    "    \"\"\"Filter based on whether a string contains a reex pattern\n",
+    "    \"\"\"\n",
+    "    df['present'] = df['EventID'].str.contains(pattern)\n",
+    "    return df[df.present == True]\n",
+    "\n",
+    "\n",
+    "def read_data(filename, **kwargs):\n",
+    "    \"\"\"\n",
+    "    \"\"\"\n",
+    "    gdf = dask_cudf.read_csv(filename, **kwargs)\n",
+    "    return gdf\n",
+    "\n",
+    "\n",
+    "def load_regex_yaml(yaml_file):\n",
+    "    with open(yaml_file) as f:\n",
+    "        regex_dict = yaml.safe_load(f)\n",
+    "        regex_dict = {k: v[0] for k, v in regex_dict.items()}\n",
+    "    return regex_dict\n",
+    "\n",
+    "\n",
+    "def create_regex_dictionaries(yaml_directory):\n",
+    "    regex_dict = {}\n",
+    "    for f in os.listdir(yaml_directory):\n",
+    "        temp_regex = load_regex_yaml(yaml_directory + '/' + f)\n",
+    "        regex_dict[f[:-5]] = temp_regex\n",
+    "        \n",
+    "    return regex_dict\n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Run Parsing Data Pipeline\n",
+    "\n",
+    "In this instance, we'll be focusing on parsing raw Windows Event Logs that are of event code [4624](https://www.ultimatewindowssecurity.com/securitylog/encyclopedia/event.aspx?eventid=4624) and [4625](https://www.ultimatewindowssecurity.com/securitylog/encyclopedia/event.aspx?eventid=4625)."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 4,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "(2515, 32)\n"
+     ]
+    }
+   ],
+   "source": [
+    "#raw lanl data parsing.\n",
+    "AUTH_INPUT_PATH = 'data/input/lanl/wls.csv'\n",
```

Comment:
Maybe above this cell, include a `!tar -xzvf /data/lanl/wls.tar.gz` and move it to this location for the user.

It's also a good idea to make the cell idempotent, so that if run twice, it doesn't error or duplicate the data.

Reply:
Sure I will update this